src/App.jsx

```diff
@@ -14,45 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_GOODS_LENGTH = 'length';
+const SORT_GOODS_ALPHABETICALLY = 'alphabetically';
+
+function sortGoods(goods, { sort }) {
+  const goodS = [...goods];
+
+  if (sort) {
+    switch (sort) {
+      case 'length':
+        goodS.sort((a, b) => a.length - b.length);
+
+        return goodS;
+
+      case 'alphabetically':
+        goodS.sort((a, b) => a.localeCompare(b));
+
+        return goodS;
+
+      default:
+        break;
+    }
+  }
+
+  return goodS;
+}
+
+export const App = () => {
+  const [sort, setSort] = useState('');
+  const [reverse, setRevers] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, { sort });
+  const booleanReset = sort !== '' || reverse;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (reverse) {
+    visibleGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button is-success',
+            { 'is-light': sort !== SORT_GOODS_ALPHABETICALLY },
+          )}
+          onClick={() => {
+            setSort(SORT_GOODS_ALPHABETICALLY);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-success', { 'is-light': sort !== SORT_GOODS_LENGTH },
```

Comment:
<img width="545" alt="Screenshot 2023-06-29 at 17 16 10" src="https://github.com/mate-academy/react_list-of-goods-js/assets/53228259/83dd4470-aaa6-4d07-8980-89d3badcd2b5">
https://offsset.github.io/react_list-of-goods-js/
check classnames